src/nerf/gun.gleam

```diff
@@ -9,12 +9,29 @@ pub external type StreamReference
 
 pub external type ConnectionPid
 
+pub type Protocols {
+  Http
+  Http2
```

Comment:
What happens if I pass both `Http` and `Http2`? Does it work?

Reply:
To my understanding yes, it will work. Or worst case gun will through an error but the docs say that its an array of Http or Http2

Reply:
Yes, I have tested that the values still work as intended. I can test again after formatting?

Reply:
What values did you test and how?

Reply:
> What combinations of values did you test and how?

I just put them into the function on connection and checked it didn't crash, couldn't think of a better test 😓

Reply:
Which combinations did you test? Did you test with both HTTP1 and HTTP2 servers?

Reply:
I believe so, but can re-test as it was a month ago

Reply:
We will need to be sure that all possible combinations work.

I don't think that a list is the right format here. `Some([Http, Http, Http, Http])` would be permitted but doesn't make sense. I think the question here is whether or not you should be permitted to specify both protocols. If the answer is yes, what is the disadvantage to always supplying them both? If none then I think we should remove this option and always use both.

Reply:
@lpil I have now added comment docs to all the parameters/options and there it explains how it can/should work.